<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Cookbook &#8212; diag_pipelines  documentation</title>
    
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Defining variables" href="general_processes.html" />
    <link rel="prev" title="Documentation for the genomics and metagenomics workflows" href="index.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body role="document">
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="cookbook">
<h1>Cookbook<a class="headerlink" href="#cookbook" title="Permalink to this headline">¶</a></h1>
<p>Routine procedures for diagnostic purposes using microbial genomics and metagenomics.</p>
<p>Workflows for epidemiology, anti-microbial resistance genotyping and virulence factors identification have been implemented using the <a class="reference external" href="http://snakemake.readthedocs.io/en/stable/">Snakemake</a> workflow management system with <a class="reference external" href="https://bioconda.github.io/">bioconda</a> integration for software dependency. <a class="reference external" href="https://hub.docker.com/r/metagenlab/diag_pipelines/">Docker</a> images of main releases are available.</p>
<div class="section" id="dependencies">
<h2>Dependencies<a class="headerlink" href="#dependencies" title="Permalink to this headline">¶</a></h2>
<p>Docker</p>
<p><code class="docutils literal"><span class="pre">docker</span> <span class="pre">pull</span> <span class="pre">metagenlab/diag_pipelines:ring_trial_v0.1.3</span></code></p>
</div>
<div class="section" id="general-use">
<h2>General use<a class="headerlink" href="#general-use" title="Permalink to this headline">¶</a></h2>
<p>Once you have pulled the docker image on your computer:</p>
<p><code class="docutils literal"><span class="pre">docker</span> <span class="pre">run</span> <span class="pre">-t</span> <span class="pre">--rm</span> <span class="pre">--mount</span> <span class="pre">source=&quot;$(pwd)&quot;,target=/home/pipeline_user/data,type=bind</span> <span class="pre">metagenlab/diag_pipelines:ring_trial_v0.1.3</span> <span class="pre">sh</span> <span class="pre">-c</span> <span class="pre">'snakemake</span> <span class="pre">--snakefile</span> <span class="pre">$pipeline_folder/workflows/ring_trial/pipeline.rules</span> <span class="pre">--use-conda</span> <span class="pre">--conda-prefix</span> <span class="pre">$conda_folder</span> <span class="pre">--configfile</span> <span class="pre">config.yaml'</span></code></p>
<p>Update the config file for your needs.</p>
</div>
<div class="section" id="generating-files-of-interest">
<h2>Generating files of interest<a class="headerlink" href="#generating-files-of-interest" title="Permalink to this headline">¶</a></h2>
<p>The pipeline works by asking the generation of the files of interest for a particular analysis. Consult the full documentation to know what files can be generated. Main examples are provided below:</p>
<p><code class="docutils literal"><span class="pre">docker</span> <span class="pre">run</span> <span class="pre">-t</span> <span class="pre">--rm</span> <span class="pre">--mount</span> <span class="pre">source=&quot;$(pwd)&quot;,target=/home/pipeline_user/data,type=bind</span> <span class="pre">metagenlab/diag_pipelines:ring_trial_v0.1.3</span> <span class="pre">sh</span> <span class="pre">-c</span> <span class="pre">'snakemake</span> <span class="pre">--snakefile</span> <span class="pre">$pipeline_folder/workflows/ring_trial/pipeline.rules</span> <span class="pre">--use-conda</span> <span class="pre">--conda-prefix</span> <span class="pre">$conda_folder</span> <span class="pre">--configfile</span> <span class="pre">config.yaml</span> <span class="pre">config.yaml</span> <span class="pre">quality/multiqc/self_genome/multiqc_report.html'</span></code></p>
<p>This will assemble and annotate every samples, and generate a multiqc report for all samples.</p>
<p><code class="docutils literal"><span class="pre">docker</span> <span class="pre">run</span> <span class="pre">-t</span> <span class="pre">--rm</span> <span class="pre">--mount</span> <span class="pre">source=&quot;$(pwd)&quot;,target=/home/pipeline_user/data,type=bind</span> <span class="pre">metagenlab/diag_pipelines:ring_trial_v0.1.3</span> <span class="pre">sh</span> <span class="pre">-c</span> <span class="pre">'snakemake</span> <span class="pre">--snakefile</span> <span class="pre">$pipeline_folder/workflows/ring_trial/pipeline.rules</span> <span class="pre">--use-conda</span> <span class="pre">--conda-prefix</span> <span class="pre">$conda_folder</span> <span class="pre">--configfile</span> <span class="pre">config.yaml</span> <span class="pre">virulence_summary.xlsx'</span></code></p>
<p>This will generate a summary excel file for the virulence factors of the samples, based on the virulence factors annotated in the file defined on the config file.</p>
<p><code class="docutils literal"><span class="pre">docker</span> <span class="pre">run</span> <span class="pre">-t</span> <span class="pre">--rm</span> <span class="pre">--mount</span> <span class="pre">source=&quot;$(pwd)&quot;,target=/home/pipeline_user/data,type=bind</span> <span class="pre">metagenlab/diag_pipelines:ring_trial_v0.1.3</span> <span class="pre">sh</span> <span class="pre">-c</span> <span class="pre">'snakemake</span> <span class="pre">--snakefile</span> <span class="pre">$pipeline_folder/workflows/ring_trial/pipeline.rules</span> <span class="pre">--use-conda</span> <span class="pre">--conda-prefix</span> <span class="pre">$conda_folder</span> <span class="pre">--configfile</span> <span class="pre">config.yaml</span> <span class="pre">typing/freebayes_joint_genotyping/core_ridom/33148/bwa/distances_snp.xlsx'</span></code></p>
<p>This will generate a snp-distance matrix of all samples, only on the core genome calculated with parsnp and with all complete genomes of the species defined in the <cite>taxid</cite> variable of the config file, mapped on the assembly (from the <a class="reference external" href="https:/www.ncbi.nlm.nih.gov/assembly/">NCBI Assembly database</a>) whose <cite>id</cite> is 33148 (<em>Staphylococcus aureus</em> COL substrain, reference genome for the ridom cgMLST scheme) with bwa</p>
<p><code class="docutils literal"><span class="pre">docker</span> <span class="pre">run</span> <span class="pre">-t</span> <span class="pre">--rm</span> <span class="pre">--mount</span> <span class="pre">source=&quot;$(pwd)&quot;,target=/home/pipeline_user/data,type=bind</span> <span class="pre">metagenlab/diag_pipelines:ring_trial_v0.1.3</span> <span class="pre">sh</span> <span class="pre">-c</span> <span class="pre">'snakemake</span> <span class="pre">--snakefile</span> <span class="pre">$pipeline_folder/workflows/ring_trial/pipeline.rules</span> <span class="pre">--use-conda</span> <span class="pre">--conda-prefix</span> <span class="pre">$conda_folder</span> <span class="pre">--configfile</span> <span class="pre">config.yaml</span> <span class="pre">typing/mlst/summary.xlsx'</span></code></p>
<p>This will generate an Excel summary file of the MLST of all samples, based on the software <a class="reference external" href="https:/github.com/tseemann/mlst">mlst</a>)</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Cookbook</a><ul>
<li><a class="reference internal" href="#dependencies">Dependencies</a></li>
<li><a class="reference internal" href="#general-use">General use</a></li>
<li><a class="reference internal" href="#generating-files-of-interest">Generating files of interest</a></li>
</ul>
</li>
</ul>
<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">Documentation for the genomics and metagenomics workflows</a></li>
      <li>Next: <a href="general_processes.html" title="next chapter">Defining variables</a></li>
  </ul></li>
</ul>
</div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/includeme.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2018, Sacha Laurent.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.5.6</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/includeme.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>